components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,183 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#ifndef MDC_SUBCLASSING_RESTRICTED
+#if defined(__has_attribute) && __has_attribute(objc_subclassing_restricted)
+#define MDC_SUBCLASSING_RESTRICTED __attribute__((objc_subclassing_restricted))
+#else
+#define MDC_SUBCLASSING_RESTRICTED
+#endif
+#endif  // #ifndef MDC_SUBCLASSING_RESTRICTED
+
+@class MDCActionSheetAction;
+
+/**
+ MDCActionSheetController displays an alert message to the user, similar to
+ UIAlertControllerStyleActionSheet.
+
+ A Material Action Sheet consists of a title, message and a list of actions.
+
+ Learn more about [Material bottom
+ sheet](https://material.io/design/components/sheets-bottom.html)
+ [Material spec list](https://material.io/design/components/lists.html)
+
+ To learn more about
+ [UIAlertController](https://developer.apple.com/documentation/uikit/uialertcontroller)
+ or [UIAlertControllerStyleActionSheet](https://developer.apple.com/documentation/uikit/uialertcontrollerstyle/uialertcontrollerstyleactionsheet)
+
+ MDCActionSheetController does not support UIPopoverController, instead it will always be presented
+ in a sheet from the bottom.
+
+ ### Dependencies
+
+ MDCActionSheetController depends on BottomSheet Component.
+ */
+MDC_SUBCLASSING_RESTRICTED
+@interface MDCActionSheetController : UIViewController
```

Comment:
Suggestion:
If we make the public API a protocol we can land this PR without the implementation.

This will make it easier to land smaller PRs and not have to wait until the component is complete before we land anything/everything at once. It also blends nicely with "use more protocols" advice out of our internal conference.